pkg/controller/controller.go

```diff
@@ -192,17 +195,20 @@ func (ctrl *resizeController) updatePVC(oldObj, newObj interface{}) {
 	// know how to support resizing of a "un-annotated" in-tree PVC. When in-tree resizer does add the annotation, a second
 	// update even will be received and we add the pvc to workqueue. If annotation matches the registered driver name in
 	// csi_resizer object, we proceeds with expansion internally or we discard the PVC.
-	// 2. An already expanded in-tree PVC:
+	// 2. A PVC that was just expanded:
+	// Once PVC's status capacity is fully updated, it is required that PV's annotation with pre-resize capacity to be removed
+	// 3. An already expanded in-tree PVC:
 	// An in-tree PVC is resized with in-tree resizer. And later, CSI migration is turned on and resizer name is updated from
 	// in-tree resizer name to CSI driver name.
-	if newSize.Cmp(oldSize) > 0 || newResizerName != oldResizerName {
+	if newReq.Cmp(oldReq) > 0 || !newCap.Equal(oldCap) || newResizerName != oldResizerName {
```

Comment:
Same comment as below for checking with equality. But I am also wondering if there is a better way. Can we check for presence of annotation on PVs that this PVC is bound to?

Reply:
Yes we can surely do that but I just thought it might be a bit redundant as we check for the annotation in `syncPVC`.